addons/mail/static/src/model/model_field_relation_set.js

```diff
@@ -49,93 +49,111 @@ export class RelationSet {
-                                    break;
-                                }
-                                case 'case-insensitive-asc': {
-                                    if (typeof valA !== 'string' || typeof valB !== 'string') {
-                                        break;
-                                    }
-                                    const cleanedValA = cleanSearchTerm(valA);
-                                    const cleanedValB = cleanSearchTerm(valB);
-                                    if (cleanedValA === cleanedValB) {
-                                        break;
-                                    }
-                                    return cleanedValA < cleanedValB ? -1 : 1;
-                                }
-                                case 'smaller-first':
-                                    if (typeof valA !== 'number' || typeof valB !== 'number') {
-                                        break;
-                                    }
-                                    if (valA === valB) {
-                                        break;
-                                    }
-                                    return valA - valB;
-                                case 'greater-first':
-                                    if (typeof valA !== 'number' || typeof valB !== 'number') {
-                                        break;
-                                    }
-                                    if (valA === valB) {
-                                        break;
-                                    }
-                                    return valB - valA;
-                                case 'most-recent-first':
-                                    if (!(valA instanceof Date) || !(valB instanceof Date)) {
-                                        break;
-                                    }
-                                    if (valA === valB) {
-                                        break;
-                                    }
-                                    return valB - valA;
-                                default:
-                                    throw Error(`sort compare method "${compareMethod}" is not supported.`);
-                            }
-                        }
-                        return 0;
-                    };
-                    // Naive method: re-sort the complete array every time. Ideally each item should
-                    // be inserted/moved at its correct place immediately, but this can be optimized
-                    // eventually if necessary.
-                    this.sortArray.sort(compareFunction);
+                    if (info.reason !== 'initial call') {
+                        // Naive method: re-sort the complete array every time. Ideally each item should
+                        // be inserted/moved at its correct place immediately, but this can be optimized
+                        // eventually if necessary.
```

Comment:
comment need to be adapted, items are correctly inserted now, but it's the update of observed fields that requires a complete sort